library/modules/Persistence.cpp

```diff
@@ -291,6 +374,54 @@ static bool load_file(const std::filesystem::path & path, int entity_id) {
     return true;
 }
 
+static bool load_persist_files(const std::filesystem::path& path)
+{
+    Json::Value json;
+    try {
+        std::ifstream file(path);
+        file >> json;
+    }
+    catch (std::exception&) {
+        // empty file?
+        return false;
+    }
+
+    size_t max_idx = 0;
+
+    if (json.isArray())
+    {
+        for (auto& entity : json)
+        {
+            if (entity.isMember("e"))
+            {
+                auto pair = file_storage.try_emplace(entity["e"].asInt());
+                if (entity.isMember("arr"))
+                {
+                    auto arr = entity["arr"];
+                    if (arr.isArray())
+                    {
+                        for (auto& k_v : arr)
+                        {
+                            if (k_v.isMember("k") && k_v.isMember("v"))
+                            {
+                                const size_t this_idx = k_v["v"].asUInt64();
+                                max_idx = std::max(max_idx, this_idx);
+
+                                pair.first->second[k_v["k"].asString()] = FileEntry{this_idx, false};
+                                //file_storage[e][k] = v
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
```

Comment:
i don't like the deep nesting here and would suggest using `std::ranges::filter` to prefilter the iterators instead